os/fs/smartfs/README.txt

```diff
@@ -1,90 +1,179 @@
-SMARTFS README
+jSmartFS README
 ^^^^^^^^^^^^^^
 
 This README file contains information about the implemenation of the TinyAra
 Sector Mapped Allocation for Really Tiny (SMART) FLASH file system, SMARTFS.
 
+The same has now been upgraded to jSmartFs with the addition of 2 key features -
+Filesystem Consistency and Two-Tier File Data Integrity.
+
 Contents:
 
-  Features
-  General operation
-  SMARTFS organization
-  Headers
-  Multiple mount points
-  SMARTFS Limitations
-  ioctls
-  Things to Do
+  > Features
+  > General operation
+  > SMARTFS organization
+  > Headers
+  > Multiple mount points
+  > SMARTFS Limitations
+  > ioctls
+  > Things to Do
 
 Features
 ========
 
   This implementation is a full-feature file system from the perspective of
-  file and directory access.  The SMART File System was designed specifically
+  file and directory access. The SMART File System was designed specifically
   for small SPI based FLASH parts (1-8 Mbyte for example), though this is not
-  a limitation.  It can certainly be used for any size FLASH and can work with
-  any MTD device by binding it with the SMART MTD layer and has been tested with
-  devices as large as 128MByte (using a 2048 byte sector size with 65534 sectors).
+  a limitation.  It can certainly be used for flash memories of any size and has
+  been tested with devices as large as 128MiB (using a 2048 byte sector size
+  with 65534 sectors).
   The FS includes support for:
+    - Support for multiple mount points on a single volume / partition.
     - Multiple open files from different threads.
-    - Open for read/write access with seek capability.
-    - Appending to end of files in either write, append or read/write
-      open modes.
-    - Directory support.
-    - Support for multiple mount points on a single volume / partition (see
-      details below).
-    - Selectable FLASH Wear leveling algorithym
-    - Selectable CRC-8 or CRC-16 error detection for sector data
+    - Directory Support
+    - Open for reading / writing with seek capability.
+    - Appending to the end of files and Overwriting existing file contents.
+    - Static Wear leveling algorithm.
+    - CRC-16 checksums for error detection in data
     - Reduced RAM model for FLASH geometries with large number of sectors (16K-64K)
-    - Recovery corrupted sector & Logging based Journaling
+    - Recovery of corrupted sectors & Logging based Journaling
+
+ Key additions to jSmartFS include:
+    - Recovery mechanism to take care of -
+	-- Isolated sectors
+	-- Broken Chains
+	-- Duplications of logical sectors
+    - Log based journaling which tracks every Block Driver Layer API which
+      makes changes to the flash contents.
+
+ With the above mentioned capabilities, jSmartFs:
+    - guarantees file system concistency, i.e. synchronization of all filesystem
+      data structures with each other so that the system remains fit to use.
+    - guarantees atomicity of block driver operations at the HAL.
```

Comment:
actually we guarantees all file operation except write. and regarding write, we guarantees operation on current sector

Reply:
Actually journaling has added to guarantees atomicity of file operation it is goal.